src/main/java/org/folio/circulation/resources/subscribers/LoanRelatedFeeFineClosedSubscriberResource.java

```diff
@@ -0,0 +1,115 @@
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.RouteRegistration;
+import org.folio.circulation.support.http.server.NoContentResponse;
+import org.folio.circulation.support.http.server.ValidationError;
+import org.folio.circulation.support.http.server.WebContext;
+
+import io.vertx.core.http.HttpClient;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+
+public class LoanRelatedFeeFineClosedSubscriberResource extends Resource {
+  private static final List<String> LOST_ITEM_FEE_TYPES = Arrays.asList(
+    LOST_ITEM_FEE_TYPE, LOST_ITEM_PROCESSING_FEE_TYPE);
+
+  public LoanRelatedFeeFineClosedSubscriberResource(HttpClient client) {
+    super(client);
+  }
+
+  @Override
+  public void register(Router router) {
+    new RouteRegistration("/circulation/subscribers/loan-related-fee-fina-closed", router)
+      .create(this::handleFeeFineClosedEvent);
+  }
+
+  private void handleFeeFineClosedEvent(RoutingContext routingContext) {
+    final WebContext context = new WebContext(routingContext);
+
+    createAndValidateRequest(routingContext)
+      .after(request -> processEvent(context, request))
+      .thenApply(r -> r.toFixedValue(NoContentResponse::noContent))
+      .thenAccept(context::writeResultToHttpResponse);
+  }
+
+  private CompletableFuture<Result<Loan>> processEvent(
+    WebContext context, LoanRelatedFeeFineClosedEvent event) {
+
+    final Clients clients = create(context, client);
+    final LoanRepository loanRepository = new LoanRepository(clients);
+
+    return loanRepository.getById(event.getLoanId())
+      .thenCompose(r -> r.after(loan -> {
+        if (loan.isDeclaredLost()) {
+          return closeDeclaredLostLoanIfLostFeesResolved(clients, loan);
+        }
+
+        return completedFuture(succeeded(loan));
+      }));
+  }
+
+  private CompletableFuture<Result<Loan>> closeDeclaredLostLoanIfLostFeesResolved(
+    Clients clients, Loan loan) {
+
+    final AccountRepository accountRepository = new AccountRepository(clients);
+    final LostItemPolicyRepository lostItemPolicyRepository = new LostItemPolicyRepository(clients);
+    final StoreLoanAndItem storeLoanAndItem = new StoreLoanAndItem(clients);
+
+    return accountRepository.findAccountsForLoan(loan)
+      .thenComposeAsync(lostItemPolicyRepository::findLostItemPolicyForLoan)
+      .thenCompose(r -> r.after(loanWithReferenceData -> {
+        if (areLostFeesClosed(loanWithReferenceData)) {
+          loanWithReferenceData.closeLoanAsLostAndPaid();
+
+          return storeLoanAndItem.updateLoanAndItemInStorage(loanWithReferenceData);
+        }
+
+        return completedFuture(succeeded(loanWithReferenceData));
+      }));
+  }
+
+  private boolean areLostFeesClosed(Loan loan) {
+    if (loan.getLostItemPolicy().hasActualCostFee()) {
+      // Actual cost fee is processed manually
+      return false;
+    }
```

Comment:
What happens if the policy has been changed since the item was lost?

Could this check for issued actual cost fees?

Reply:
There might be no fee/fine for actual cost. Event if there is, we won't know the type name for actual cost fees/fines.

Reply:
What does the story say about this situation?

I thought it had been agreed that we weren't trying to implement the actual cost fee process  yet in part because we couldn't identify the charged fees.

Reply:
![image](https://user-images.githubusercontent.com/9094650/82663323-1a34f000-9c38-11ea-9167-62858a8d8318.png)

Reply:
Please can this behaviour be confirmed with Holly and Cate.

If that is expected, please can this be converted from a note in the story to a scenario.

Reply:
Confirmed by Emma.